revdep/problems.md

```diff
@@ -87,6 +301,122 @@ Run `revdepcheck::cloud_details(, "DiagrammeR")` for more info
       Note: found 1 marked UTF-8 string
     ```
 
+# dosearch
+
+<details>
+
+* Version: 1.0.8
+* GitHub: NA
+* Source code: https://github.com/cran/dosearch
+* Date/Publication: 2021-08-19 16:40:02 UTC
+* Number of recursive dependencies: 18
+
+Run `revdepcheck::cloud_details(, "dosearch")` for more info
+
+</details>
+
+## Newly broken
+
+*   checking examples ... ERROR
+    ```
+    Running examples in ‘dosearch-Ex.R’ failed
+    The error most likely occurred in:
+    
+    > ### Name: dosearch
+    > ### Title: Identify a causal effect from arbitrary experiments and
+    > ###   observations
+    > ### Aliases: dosearch
+    > 
+    > ### ** Examples
+    > 
+    ...
+    +   g_igraph <- igraph::set.edge.attribute(g_igraph, "description", 3:4, "U")
+    +   dosearch(data2, query2, g_igraph)
+    + }
+    Warning: `graph.formula()` was deprecated in igraph 2.0.0.
+    ℹ Please use `graph_from_literal()` instead.
+    Warning: `set.edge.attribute()` was deprecated in igraph 2.0.0.
+    ℹ Please use `set_edge_attr()` instead.
+    Error in graph_split[[x]] : subscript out of bounds
```

Comment:
?

Reply:
:thinking: 

``` r
library("igraph")
#> 
#> Attaching package: 'igraph'
#> The following objects are masked from 'package:stats':
#> 
#>     decompose, spectrum
#> The following object is masked from 'package:base':
#> 
#>     union

withr::local_seed(42)
graph.formula(x -+ z, z -+ y, x -+ y, y -+ x) |> 
  set.edge.attribute("description", 3:4, "U")
#> Warning: `set.edge.attribute()` was deprecated in igraph 2.0.0.
#> ℹ Please use `set_edge_attr()` instead.
#> This warning is displayed once every 8 hours.
#> Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
#> generated.
#> Warning: `graph.formula()` was deprecated in igraph 2.0.0.
#> ℹ Please use `graph_from_literal()` instead.
#> This warning is displayed once every 8 hours.
#> Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
#> generated.
#> IGRAPH 64864e9 UN-- 4 0 -- 
#> + attr: name (v/c), description (e/c)
#> + edges from 64864e9 (vertex names):

withr::local_seed(42)
graph_from_literal(x -+ z, z -+ y, x -+ y, y -+ x) |> 
  set_edge_attr("description", 3:4, "U")
#> IGRAPH 0d4414a DN-- 3 4 -- 
#> + attr: name (v/c), description (e/c)
#> + edges from 0d4414a (vertex names):
#> [1] x->z x->y z->y y->x
```

<sup>Created on 2024-01-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Ah! it's due to not passing a string!

``` r
library("igraph")
#> 
#> Attaching package: 'igraph'
#> The following objects are masked from 'package:stats':
#> 
#>     decompose, spectrum
#> The following object is masked from 'package:base':
#> 
#>     union

withr::local_seed(42)
graph.formula(x -+ z, z -+ y, x -+ y, y -+ x) 
#> Warning: `graph.formula()` was deprecated in igraph 2.0.0.
#> ℹ Please use `graph_from_literal()` instead.
#> This warning is displayed once every 8 hours.
#> Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
#> generated.
#> IGRAPH 6123ba1 UN-- 4 0 -- 
#> + attr: name (v/c)
#> + edges from 6123ba1 (vertex names):
graph.formula("x -+ z, z -+ y, x -+ y, y -+ x") 
#> IGRAPH 13e2d98 UN-- 1 0 -- 
#> + attr: name (v/c)
#> + edges from 13e2d98 (vertex names):

withr::local_seed(42)
graph_from_literal(x -+ z, z -+ y, x -+ y, y -+ x)
#> IGRAPH 6675e04 DN-- 3 4 -- 
#> + attr: name (v/c)
#> + edges from 6675e04 (vertex names):
#> [1] x->z x->y z->y y->x
graph_from_literal("x -+ z, z -+ y, x -+ y, y -+ x")
#> IGRAPH 4a23f1e UN-- 1 0 -- 
#> + attr: name (v/c)
#> + edges from 4a23f1e (vertex names):
```

<sup>Created on 2024-01-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
I'm not too sure how to handle this. I see `lazy_dots()` might be useful here, but I couldn't get it to work. In any case we can't simply pass the ... from graph.formula() to graph_from_literal().

Reply:
https://github.com/maelle/rigraph/pull/6